ttnn/cpp/ttnn/operations/data_movement/untilize_with_halo_v2/device/kernels/dataflow/halo_gather.cpp

```diff
@@ -70,15 +71,23 @@ void copy_sticks_async(
             uint16_t nsticks = config_data[i + j + 2];
             uint32_t size = nsticks * stick_nbytes;
             uint32_t dst_offset = dst_local_idx * stick_nbytes;
-            uint32_t src_offset = src_local_idx * stick_nbytes;
+            uint32_t src_offset = src_local_idx * input_aligned_page_size;
             if constexpr (is_read) {
                 uint32_t dst_addr = out_base_l1_addr + dst_offset;
                 uint64_t src_addr = base_addr + src_offset;
-                noc_async_read(src_addr, dst_addr, size);
+                for(uint16_t k = 0; k < nsticks; k++) {
+                    noc_async_read(src_addr, dst_addr, stick_nbytes);
```

Comment:
No longer coalescing? I think we should have both versions since coalesced version is much higher performing.

Reply:
Done.